src/youtube/js/youtube.js

```diff
@@ -0,0 +1,374 @@
+    };
+
+    /**
+     * Triggers a pause on the video element.
+     * @public
+     * @method OoyalaYoutubeVideoWrapper#pause
+     */
+    this.pause = function(){
+      youtubePlayer.pauseVideo();
+      clearInterval(timeUpdateInterval);
+      timeUpdateInterval = null;
+    };
+    
+    /**
+     * Triggers a seek on the video element.
+     * @public
+     * @method OoyalaYoutubeVideoWrapper#seek
+     * @param {number} time The time to seek the video to (in seconds)
+     */
+    this.seek = function(time) {
+      if(playerReady)
+      {
+        youtubePlayer.seekTo(time,true);
+        this.controller.notify( this.controller.EVENTS.SEEKED);
+      }
+      else
+      {
+        javascriptCommandQueue.push(["seek", time]);
+      }
+    };
+
+    /**
+     * Triggers a volume change on the video element.
+     * @public
+     * @method OoyalaYoutubeVideoWrapper#setVolume
+     * @param {number} volume A number between 0 and 1 indicating the desired volume percentage
+     */
+    this.setVolume = function(volume) {
+      youtubePlayer.setVolume(volume*100);
+      this.controller.notify( this.controller.EVENTS.VOLUME_CHANGE, { "volume" :volume});
+    };
+
+    /**
+     * Sets the initial time of the video playback.
+     * @public
+     * @method OoyalaYoutubeVideoWrapper#setInitialTime
+     * @param {number} initialTime The initial time of the video (seconds)
+     */
+    this.setInitialTime = function(initialTime) {
+      if (!hasPlayed || videoEnded) {
+        this.seek(initialTime);
+      }
+    };
+
+    /**
+     * Sets the url of the video.
+     * @public
+     * @method OoyalaYoutubeVideoWrapper#setVideoUrl
+     * @param {string} youtubeId The youtube Id of the video that needs to be played. 
+     * @param {string} encoding The encoding of video stream, possible values are found in OO.VIDEO.ENCODING
+     * @returns {boolean} True or false indicating success
+     */
+    this.setVideoUrl = function(youtubeId, encoding, isLive) {   
+      if(youtubeId)
+      {
+        youtubeID = youtubeId;                
+        return true;
+      }
+      return false;
+    };
+
+    /**
+     * Sets the stream to play back based on given stream ID. Plugin must support the
+     * BITRATE_CONTROL feature to have this method called.
+     * @public
+     * @method OoyalaYoutubeVideoWrapper#setBitrate
+     * @param {string} suggestedQuality The ID of the stream to switch to. This ID will be the ID property from one
+     *   of the stream objects passed with the BITRATES_AVAILABLE VTC event.
+     *   An ID of 'auto' should return the plugin to automatic bitrate selection.
+     */
+    this.setBitrate = function(suggestedQuality) {
+      youtubePlayer.setPlaybackQuality(suggestedQuality);
+    };
+
+    /**
+     * Destroys the individual video element.
+     * @public
+     * @method OoyalaYoutubeVideoWrapper#destroy
+     */
+    this.destroy = function() {
+      // Reset the source
+      this.setVideoUrl('');
```

Comment:
we had issues on Edge on Windows with setting the video to "" on destroy. It would cause the browser to try and load the main page again, messing up page analytics for the customer.  Can you put in a "isEdge" check and then not reset the videoUrl.